warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/VirtualIngest.java

```diff
@@ -145,59 +148,73 @@ public void setup(Type type, String instance, Configuration config) {
+                    throw new IllegalArgumentException("Virtual field names and members do not match. Fix configuration and try again.");
 
-            if (missingPolicies.length != 0 && missingPolicies.length != 1 && missingPolicies.length != fieldNames.length)
-                throw new IllegalArgumentException(
-                                "Virtual field names and missing policies do not match.  Specify 0, 1, or the same number of policies. Fix configuration and try again.");
+                if (groupingPolicies.length != 0 && groupingPolicies.length != 1 && groupingPolicies.length != fieldNames.length)
+                    throw new IllegalArgumentException(
+                                    "Virtual field names and grouping policies do not match.  Specify 0, 1, or the same number of policies. Fix configuration and try again.");
 
-            if (fieldNames.length == 0)
-                return;
+                if (missingPolicies.length != 0 && missingPolicies.length != 1 && missingPolicies.length != fieldNames.length)
+                    throw new IllegalArgumentException(
+                                    "Virtual field names and missing policies do not match.  Specify 0, 1, or the same number of policies. Fix configuration and try again.");
 
-            if (StringUtils.isEmpty(defaultSeparator) && StringUtils.isEmpty(defaultStartSeparator) && StringUtils.isEmpty(defaultEndSeparator))
-                throw new IllegalArgumentException(getConfPrefixes(type, instance)[0] + VIRTUAL_FIELD_VALUE_SEPARATOR + " must be specified.");
+                if (fieldNames.length == 0)
+                    return;
 
-            if (StringUtils.isEmpty(defaultStartSeparator) && StringUtils.isEmpty(defaultEndSeparator)) {
-                this.defaultStartSeparator = this.defaultSeparator;
-                this.defaultEndSeparator = "";
-            }
+                for (int i = 0; i < fieldNames.length; i++) {
+                    String name = fieldNames[i];
 
-            for (int i = 0; i < fieldNames.length; i++) {
-                String name = fieldNames[i];
+                    String[] members = datawave.util.StringUtils.split(fieldMembers[i], '.');
+                    virtualFieldDefinitions.put(name, members);
 
-                String[] members = datawave.util.StringUtils.split(fieldMembers[i], '.');
-                virtualFieldDefinitions.put(name, members);
+                    if (groupingPolicies.length == 0) {
+                        grouping.put(name, DEFAULT_GROUPING_POLICY);
+                    } else if (groupingPolicies.length == 1) {
+                        grouping.put(name, GroupingPolicy.valueOf(groupingPolicies[0]));
+                    } else {
+                        grouping.put(name, GroupingPolicy.valueOf(groupingPolicies[i]));
+                    }
 
-                if (groupingPolicies.length == 0) {
-                    grouping.put(name, DEFAULT_GROUPING_POLICY);
```

Comment:
Not a fan of storing the default for every field.  This will get expensive.  Why not assume default if not in the map...